modules/backup-s3/client.go

```diff
@@ -122,24 +122,44 @@ func (s *s3Client) AllBackups(ctx context.Context,
 			Prefix:    s.config.BackupPath,
 		},
 	)
+
 	for info := range objectsInfo {
-		if strings.Contains(info.Key, ubak.GlobalBackupFile) {
-			obj, err := s.client.GetObject(ctx,
-				s.config.Bucket, info.Key, minio.GetObjectOptions{})
-			if err != nil {
-				return nil, fmt.Errorf("get object %q: %w", info.Key, err)
-			}
-			contents, err := io.ReadAll(obj)
-			if err != nil {
-				return nil, fmt.Errorf("read object %q: %w", info.Key, err)
-			}
-			var desc backup.DistributedBackupDescriptor
-			if err := json.Unmarshal(contents, &desc); err != nil {
-				return nil, fmt.Errorf("unmarshal object %q: %w", info.Key, err)
-			}
-			meta = append(meta, &desc)
+		if err := ctx.Err(); err != nil {
+			return nil, fmt.Errorf("context cancelled: %w", err)
+		}
+
+		// Only process global backup files - this is the key filter
+		// This filters out all other files and only processes backup_config.json
+		if !strings.HasSuffix(info.Key, ubak.GlobalBackupFile) {
+			continue
 		}
+
+		// Get the backup object
+		obj, err := s.client.GetObject(ctx,
+			s.config.Bucket, info.Key, minio.GetObjectOptions{})
+		if err != nil {
+			return nil, fmt.Errorf("get object %q: %w", info.Key, err)
+		}
+
+		// Ensure object is closed to prevent connection leaks
+		defer obj.Close()
+
+		// Use a buffer to limit memory usage and add timeout protection
```

Comment:
I'm curious about the effect of `io.Copy` on timeout protection here -- could you elaborate on this please?